File: Data Structures & Algorithms/partition-labels/submission-1.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        last_indexes = {}
        for i,c in enumerate(s):
            last_indexes[c] = i
        
        res = []
        start = end = 0
        for i,c in enumerate(s):
            end = max(end, last_indexes[c])
            if i == end:
                res.append(end-start+1)
                start = end+1
            
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int lowest = prices[0];

        for (int price : prices){
            lowest = min(lowest, price);
            res = max(res, price-lowest);
        }

        return res;
    }
};
